src/v/datalake/translation/deps.cc

```diff
@@ -361,14 +361,20 @@ class partition_data_source : public data_source {
 
     ss::future<std::optional<model::record_batch_reader>>
     make_log_reader(kafka::offset begin_offset, ss::abort_source& as) final {
+        // Bump the reader start offset up to the log start. Untranslated data
+        // should typically be pinned, so the input offset should be above the
+        // log start in normal circumstances, but it's possible that e.g. data
+        // was removed when Iceberg was disabled.
+        auto reader_start_offset = std::max(
+          begin_offset, model::offset_cast(_partition_proxy->start_offset()));
```

Comment:
In wasm we use `sync_effective_start` which seems to be what we validate against for the fetch handler: https://github.com/redpanda-data/redpanda/blob/c711995fb2ffe12ca82c596db53806bc4705f1d5/src/v/kafka/data/replicated_partition.cc#L642

Reply:
I guess the only difference is the log eviction STM

Reply:
Looking a bit, I think the key difference is only that `sync_effective_start()` calls `sync()` on the STMs to ensure any replicated start offset changes from previous terms are reflected in the current call, and to ensure we're still leader. We still do respect the Kafka start offset override from the log eviction STM though.

I'm inclined to keep it as is since syncing the STM doesn't feel like a hard requirement here